woocommerce/woocommerce-gutenberg-products-block
# Build: Split out vendor files into separate, shared JS & CSS files

This PR uses webpack's `splitChunks` to pull out all node_modules code into a separate vendors.js & vendors.css file, which avoids shipping `react-world-flags` in every JS file (see https://github.com/woocommerce/wc-admin/issues/1275 ). This brings down each JS file from 6.5+MB to about 25KB, and adds one large vendors.js file.

Incidentally this also seems to bring down the build time by about 2s (from 10 to 8), though it's still longer than I would want.

Before:
```
                       Asset      Size  Chunks                    Chunk Names
        featured-product.css  9.07 KiB       0  [emitted]         featured-product
         featured-product.js  6.52 MiB       0  [emitted]  [big]  featured-product
     handpicked-products.css  6.05 KiB       1  [emitted]         handpicked-products
      handpicked-products.js  6.53 MiB       1  [emitted]  [big]  handpicked-products
    product-best-sellers.css  6.49 KiB       2  [emitted]         product-best-sellers
     product-best-sellers.js  6.53 MiB       2  [emitted]  [big]  product-best-sellers
        product-category.css  7.08 KiB       3  [emitted]         product-category
         product-category.js  6.53 MiB       3  [emitted]  [big]  product-category
             product-new.css  6.49 KiB       4  [emitted]         product-new
              product-new.js  6.53 MiB       4  [emitted]  [big]  product-new
         product-on-sale.css  6.49 KiB       5  [emitted]         product-on-sale
          product-on-sale.js  6.53 MiB       5  [emitted]  [big]  product-on-sale
       product-top-rated.css  6.49 KiB       6  [emitted]         product-top-rated
        product-top-rated.js  6.53 MiB       6  [emitted]  [big]  product-top-rated
          products-block.css  16.3 KiB       7  [emitted]         products-block
           products-block.js  45.3 KiB       7  [emitted]         products-block
           products-grid.css  74.4 KiB       8  [emitted]         products-grid
            products-grid.js  5.94 KiB       8  [emitted]         products-grid
```

After:
```
                       Asset      Size  Chunks                    Chunk Names
        featured-product.css  9.07 KiB       1  [emitted]         featured-product
         featured-product.js  23.5 KiB       1  [emitted]         featured-product
     handpicked-products.css  6.05 KiB       2  [emitted]         handpicked-products
      handpicked-products.js  26.4 KiB       2  [emitted]         handpicked-products
    product-best-sellers.css  6.49 KiB       3  [emitted]         product-best-sellers
     product-best-sellers.js    26 KiB       3  [emitted]         product-best-sellers
        product-category.css  7.08 KiB       4  [emitted]         product-category
         product-category.js  31.9 KiB       4  [emitted]         product-category
             product-new.css  6.49 KiB       5  [emitted]         product-new
              product-new.js  25.8 KiB       5  [emitted]         product-new
         product-on-sale.css  6.49 KiB       6  [emitted]         product-on-sale
          product-on-sale.js  27.2 KiB       6  [emitted]         product-on-sale
       product-top-rated.css  6.49 KiB       7  [emitted]         product-top-rated
        product-top-rated.js  25.8 KiB       7  [emitted]         product-top-rated
          products-block.css  16.3 KiB       8  [emitted]         products-block
           products-block.js    38 KiB       8  [emitted]         products-block
           products-grid.css  1.39 KiB       9  [emitted]         products-grid
            products-grid.js  1.66 KiB       9  [emitted]         products-grid
                 vendors.css    73 KiB       0  [emitted]         vendors
                  vendors.js   6.5 MiB       0  [emitted]  [big]  vendors
```

### How to test the changes in this Pull Request:

1. Build the project
2. Try using the blocks
3. Expect: You should see a vendors.js file loaded in wp-admin
4. Expect: No functionality/style changes, everything should work as expected.
5. Publish a post with a Featured Product block, view it
6. Expect: The correct styles are loaded.
